pyfar/dsp/dsp.py

```diff
@@ -187,3 +187,112 @@ def spectrogram(signal, dB=True, log_prefix=20, log_reference=1,
     times -= times[0]
 
     return frequencies, times, spectrogram
+
+
+def normalize(signal, norm_type='time', operation='max', channelwise='max',
+              value=None, freq_range=(20, 22e3)):
+    """
+    Normalize signal in the time or frequency domain.
+
+    Parameters
+    ----------
+    signal: Signal
+        Input signal of the signal class
+    norm_type: string
+        'time' - Normalize the time signal to 'value'
+        'abs' - Normalize the magnitude spectrum to 'value'
+        'dB' - Normalize the db magnitude spectrum to 'value'
+        The default is 'time'
+    operation: string
+        'max' - Normalize to the absolute maximum of the signal data
+        'mean' - Normalize to the mean of the signal data
+        'rms' - Normalize to the rms of the signal data
+        The default is 'max'
+    channelwise: string
+        'each' - Normalize each channel separately
+        'max' - Normalize to the max or operation across all channels
+        'min' - Normalize to min or operation across all channels
+        'mean' - Normalize to mean of operation across all channels
+       The default is 'max'
+    value:
+        Normalizes to 'value' which can be a scalar or a vector with
+        a number of elements equal to channels
+        The default is 0 for norm_type='db' and 1 otherwise
+    freq_range: tuple
+        two element vector specifying upper and lower frequency bounds
+        for normalization or scalar specifying the centre frequency for
+        normalization
+        The default is (20,22000)
+    Returns
+    --------
+    normalized_signal: Signal
+        The normalized signal
+    """
+
+    # check input
+    if not isinstance(signal, Signal):
+        raise TypeError('Input data has to be of type: Signal.')
+
+    # set default values
+    if value is None:
+        value = 0 if norm_type == 'dB' else 1
+
+    # copy and transform data to the desired domain
+    if norm_type == 'time':
+        normalized_input = np.abs(signal.time.copy())
+    elif norm_type == 'abs':
+        normalized_input = np.abs(signal.freq.copy())
+    elif norm_type == 'dB':
+        normalized_input = 20 * np.log10(signal.freq.copy())
+    else:
+        raise ValueError(("norm _type must be 'time', 'abs' or 'dB'."))
```

Comment:
To make the code easier to read, you could name this `input` and leave the `normalized` for later